### PR TITLE
Give version source files a meaningful name (bug 1140716)

### PR DESCRIPTION
**Pull request by wagnerand**
@magopian r?

### CODE REVIEW

File: apps/versions/models.py [hunk 1]
```diff
@@ -48,10 +48,23 @@ def get_query_set(self):
 
 
 def source_upload_path(instance, filename):
+    VALID_SOURCE_EXTENSIONS = (
```
Comment (magopian) [inline apps/versions/models.py]: This needs to be factorized with https://github.com/mozilla/olympia/blob/master/apps/devhub/forms.py#L359

What I'd do: keep this one, in this file, but move it to the top below the imports, at the module level. Then import it in `apps.devhub.forms`.

File: apps/versions/models.py [hunk 2]
```diff
@@ -48,10 +48,23 @@ def get_query_set(self):
...
+        '.zip', '.tar', '.7z', '.tar.gz', '.tgz', '.tbz', '.txz', '.tar.bz2',
+        '.tar.xz'
+    )
+    file_ext = ''
+    for ext in VALID_SOURCE_EXTENSIONS:
+        if filename.endswith(ext):
+            file_ext = ext
```
Comment (magopian) [inline apps/versions/models.py]: You actually don't need to do this. When you break (or end) a loop, `ext` will be assigned to its latest value. So if you remove `file_ext = ''` and remove the affectation on line 58, you can simply use `ext` on line 67.

**Comment by magopian** [comment]
That's looking good, thanks for this contribution!

r+wc

File: apps/devhub/forms.py [hunk 3]
```diff
@@ -30,7 +30,8 @@
 from translations.fields import TransTextarea, TransField
 from translations.models import delete_translation, Translation
 from translations.forms import TranslationFormMixin
-from versions.models import License, Version, ApplicationsVersions
+from versions.models import (License, Version, ApplicationsVersions,
+                             VALID_SOURCE_EXTENSIONS)
```
Comment (magopian) [inline apps/devhub/forms.py]: I know it wasn't in the correct order previously, but please reorder the imports alphabetically while you're at it

Comment (wagnerand) [inline apps/devhub/forms.py]: Done.

File: apps/versions/models.py [hunk 4]
```diff
@@ -48,10 +53,17 @@ def get_query_set(self):
 
 
 def source_upload_path(instance, filename):
+    for ext in VALID_SOURCE_EXTENSIONS:
```
Comment (magopian) [inline apps/versions/models.py]: Please add a comment above to explain that piece of code, for clarity. Something like:
`The source file should only have been saved if its extension was in VALID_SOURCE_EXTENSIONS.`

Comment (wagnerand) [inline apps/versions/models.py]: Done.

File: apps/versions/models.py [hunk 5]
```diff
@@ -48,10 +53,20 @@ def get_query_set(self):
 
 
 def source_upload_path(instance, filename):
+    # At this point we already know that ext is one of VALID_SOURCE_EXTENSIONS
+    # because we already checked for that in
+    # /apps/devhub/forms.py#WithSourceMixin.clean_source
```
Comment (magopian) [inline apps/versions/models.py]: :+1:

**Comment by magopian** [comment]
Thanks a lot for this work @wagnerand !

**Pull request merged**